mollie/commercetools
# Feat/cmi 107 return 405 on non post request

## Description

Ticket: https://anddigitaltransformation.atlassian.net/browse/CMI-107

Calling the main endpoint with a method that isn't `POST` should return `405 - Method Not Allowed`

## Type of change


## Testing

To test locally - run the notification / extension module and make a call to the `localhost:3000/` endpoint. If it is `POST` it should process, if it's any other method (e.g. `PATCH`) it should return `405`. 

With the log level set at `http` (or lower) then this should print to the console. 
- set `LOG_LEVEL=http` and see these warnings
- set `LOG_LEVEL=warn` and you should not see these warnings

Check you can still call `GET /health` and receive 200, not a 4xx.

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works/doesn't break everything
- [x] Existing tests pass locally with my changes
